File: python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
def find_armstrong_numbers(numbers):
    output = []
    for number in numbers:
        num_digits = len([num for num in str(number)])
        sum_powers = sum(pow(number, num_digits) for number in (int(num) for num in str(number)))
        if sum_powers == number:
            output.append(number)
    return output

File: js/armstrongNumbers.js
// How can you make this more scalable and reusable later?

function findArmstrongNumbers(numbers) {
  let output = [];
  for (number in numbers) {
    let digits = number.toString().split("");
    const quantity_digits = digits.length;
    let arr_num = digits.map(Number);
    let sum = 0;
    for (digit of arr_num) {
      let pow = Math.pow(digit, quantity_digits);
      sum += pow;
    }
    if (sum == number) {
      output.push(Number(number));
    }
  }
  return output;
}

module.exports = findArmstrongNumbers;